packages/client/hmi-client/src/components/drilldown-panel/selected-resources-options-pane.vue

```diff
@@ -131,6 +138,17 @@ const addResourcesToProject = async (projectId: string) => {
 			validProject.value?.assets.models.push(modelId);
 			resources.activeProjectAssets?.[MODELS].push(selectedItem);
 		}
+		if (isDataset(selectedItem)) {
+			// FIXME: handle cases where assets is already added to the project
+			const datasetId = selectedItem.id;
+			// then, link and store in the project assets
+			const assetsType = DATASETS;
+			await ProjectService.addAsset(projectId, assetsType, datasetId);
+
+			// update local copy of project assets
+			validProject.value?.assets.datasets.push(datasetId);
+			resources.activeProjectAssets?.[DATASETS].push(selectedItem);
```

Comment:
This should be handled in the Service Layer. In `addAsset` to avoid handling changing multiple source of truth at different places.

Reply:
This is part of the refactoring planned after Dec 8